### PR TITLE
Feature/#11515 continuous searching

### PR DESCRIPTION
**Pull request by ovidiuvladpop**

### CODE REVIEW

File: ClearentIdtechIOSFramework/ClearentUI/ClearentProcessingModalPresenter.swift [hunk 1]
```diff
@@ -10,6 +10,7 @@ import UIKit
 
 protocol ClearentProcessingModalView: AnyObject {
     func updateContent(with feedback: FlowFeedback)
+    func updateCurrentContentWithLoadingView()
```
[the remark this answers is not in the document]
Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/ClearentUI/ClearentProcessingModalPresenter.swift]: Updated

File: ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift [hunk 2]
```diff
@@ -77,6 +76,10 @@ public final class ClearentWrapper : NSObject {
     private let monitor = NWPathMonitor()
     private var isInternetOn = false
     internal var isBluetoothOn = false
+    private var continuousSearchingTimer: Timer?
+    private var shouldStopUpdatingReadersListDuringContinuousSearching: Bool = true
+    private var shouldAddLoadingViewDuringContinuousSearching: Bool = false
+    public var shouldBeginContinuousSearchingForReaders: ((_ shouldStart: Bool) -> Void)?
```
[the remark this answers is not in the document]
Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift]: Updated

File: ClearentIdtechIOSFramework/ClearentUI/Common/Util/ClearentWrapperDefaults.swift [hunk 3]
```diff
@@ -73,6 +74,15 @@ extension ClearentWrapperDefaults {
         }
         set {
             ClearentWrapper.shared.readerInfoReceived?(newValue)
+            // current paired reader should be in sync with the corresponding item from previously paired readers
+            if let pairedReader = newValue {
+                ClearentWrapper.shared.addReaderToRecentlyUsed(reader: pairedReader)
+            } // if pairedReader is nil, we need to set connected and autojoin to false in the previously paired readers
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentUI/Common/Util/ClearentWrapperDefaults.swift:80]:
```suggestion
            } // if pairedReader is nil, we need to set connected and autojoin to false for the corresponding item from previously paired readers
```

File: ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift [hunk 4]
```diff
@@ -158,13 +172,13 @@ public final class ClearentWrapper : NSObject {
...
-            
-            strongSelf.searchingRecentlyUsedReadersInProgress = true
-            strongSelf.clearentVP3300.start(ClearentConnection(bluetoothSearch: ()))
+        let usedReaders = ClearentWrapperDefaults.recentlyPairedReaders
+        if (usedReaders?.count == 0) {
+            self.delegate?.didNotFindRecentlyUsedReaders()
+        } else {
+            if let usedReaders = usedReaders {
+                self.delegate?.didFindRecentlyUsedReaders(readers: usedReaders)
+            }
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift:181]:
```suggestion
        if let usedReaders = ClearentWrapperDefaults.recentlyPairedReaders, usedReaders.count > 0 {
             delegate?.didFindRecentlyUsedReaders(readers: usedReaders)
        } else {
             delegate?.didNotFindRecentlyUsedReaders()
        }
```

File: ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift [hunk 5]
```diff
@@ -414,41 +428,32 @@ extension ClearentWrapper : Clearent_Public_IDTech_VP3300_Delegate {
...
-                }
-            } else {
-                let readers = fetchRecentlyAndAvailableReaders(devices: bluetoothDevices)
-                self.delegate?.didFindRecentlyUsedReaders(readers: readers)
-            }
-        } else if (bluetoothDevices.count > 0) {
+        if (bluetoothDevices.count > 0) {
             let readers = bluetoothDevices.map { device in
                 return readerInfo(from: device)
             }
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift:434]:
```suggestion
            let readers = bluetoothDevices.map { readerInfo(from: $0) }
```

File: ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift (hunk 4 above)
Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift]: Updated

File: ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift (hunk 5 above)
Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/Wrapper/ClearentWrapper.swift]: Updated

File: ClearentIdtechIOSFramework/ClearentUI/Common/Util/ClearentWrapperDefaults.swift (hunk 3 above)
Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/ClearentUI/Common/Util/ClearentWrapperDefaults.swift]: Updated

**Review by carmenj — APPROVED** [review]

**Pull request merged**